sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGetConfig.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Input for the BatchGet operation in the object-persistence programming model
+    /// </summary>
+#if NET8_0_OR_GREATER
+    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Custom.Internal.InternalConstants.RequiresUnreferencedCodeMessage)]
+#endif
+    public class BatchGetConfig
+    {
+        /// <summary>
+        /// Property that indicates which DynamoDB table to use, 
+        /// overriding the DynamoDBTable attribute declared for the object type.
+        /// </summary>
+        public string OverrideTableName { get; set; }
+
+        /// <summary>
+        /// Property that directs DynamoDBContext to prefix all table names
+        /// with a specific string.
+        /// If property is null or empty, no prefix is used and default
+        /// table names are used.
+        /// </summary>
+        public string TableNamePrefix { get; set; }
+
+        /// <summary>
+        /// The object persistence model API relies on an internal cache of the DynamoDB table's metadata to construct and validate 
+        /// requests. This controls how the cache key is derived, which influences when the SDK will call 
+        /// IAmazonDynamoDB.DescribeTable(string) internally to populate the cache.
```

Comment:
nit: add doc reference to `IAmazonDynamoDB.DescribeTable(string)`

Reply:
`DescribeTable` isn't available in all targets, it's `DescribeTableAync` in .NET/Core/Standard.

I didn't think it was worth `#if` in the doc string, and the API reference generator is going to pick one anyway.

Reply:
Resolved over in #3380